lib/context.js

```diff
@@ -266,6 +266,53 @@ api.process = async ({
       defined.set('@propagate', true);
     }
 
+    // handle @import
+    if('@import' in ctx) {
+      const value = ctx['@import'];
+      if(activeCtx.processingMode === 'json-ld-1.0') {
+        throw new JsonLdError(
+          'Invalid JSON-LD syntax; @import not compatible with ' +
+          activeCtx.processingMode,
+          'jsonld.SyntaxError',
+          {code: 'invalid context member', context: ctx});
+      }
+      if(!_isString(value)) {
+        throw new JsonLdError(
+          'Invalid JSON-LD syntax; @import must be a string.',
+          'jsonld.SyntaxError',
+          {code: 'invalid @import value', context: localCtx});
+      }
+
+      // resolve contexts
+      const resolvedImport = await options.contextResolver.resolve({
+        context: value,
+        documentLoader: options.documentLoader,
+        base: options.base
+      });
+      if(resolvedImport.length !== 1) {
+        throw new JsonLdError(
+          'Invalid JSON-LD syntax; @import must reference a single context.',
+          'jsonld.SyntaxError',
+          {code: 'invalid remote context', context: localCtx});
+      }
+      const importCtx = resolvedImport[0].document;
+      if('@import' in importCtx) {
```

Comment:
Seems like this would need to be more involved to catch `@import` in scoped contexts too. There are not test for this.

Reply:
Catching it was going to require adding more state to be passed down, and even though it isn't the use case, it doesn't really cause any harm if its (mis)used this way. But, that would be an issue on the spec.